[Mobile]: More functionalities

joel-jeremy opened the pull request:
<!-- Thank you for submitting a pull request! Make sure to follow the instructions to write release notes for your PR — it should only take a minute or two: https://github.com/actualbudget/docs#writing-good-release-notes -->
Added for mobile:
- Updated the balance tooltip to be a modal
- Scheduled transaction action modals
  - Post transaction
  - Skip scheduled date
  - View schedule (not supported for now)
- Budget summary action modals
  - Move to a category
  - Hold for next month
  - Reset hold buffer
- Account action modals
  - Rename account
  - Close account
  - Reopen account

MatissJanis commented:
This is a massive amount of LOC. I started reviewing, but quickly got lost. Any chance you could break this out in smaller PRs?

joel-jeremy commented:
> This is a massive amount of LOC. I started reviewing, but quickly got lost. Any chance you could break this out in smaller PRs?

Sorry for the big PR. I had some extra time so I tried doing as much as I can before I get a bit busy again. Any tips how to break this down into smaller chunks?

For some of the changes, I have:
- Extracted some `Menu` components used in desktop e.g. BalanceMenu, ToBudgetMenu to their own files so that they can be reused in mobile.
- Made some changes to the Autocomplete component to fix the typecheck errors that I was encountering.
- Extracted category/payee/account autocomplete modals from EditField to their own files/components

joel-jeremy commented on packages/desktop-client/src/components/Modals.tsx:
```diff
@@ -281,17 +335,22 @@ export function Modals() {
         case 'rollover-budget-summary':
           return (
-            <RolloverBudgetSummary
+            <NamespaceContext.Provider
```
We need to think about how to remove the need to wrap modal components with this in order for `useSheetValue` to work. By default, there is no namespace context for any of our modals.

MatissJanis commented:
> > This is a massive amount of LOC. I started reviewing, but quickly got lost. Any chance you could break this out in smaller PRs?
> 
> Sorry for the big PR. I had some extra time so I tried doing as much as I can before I get a bit busy again. Any tips how to break this down into smaller chunks?
> 
> For some of the changes, I have:
> 
> * Extracted some `Menu` components used in desktop e.g. BalanceMenu, ToBudgetMenu to their own files so that they can be reused in mobile.
> * Made some changes to the Autocomplete component to fix the typecheck errors that I was encountering.
> * Extracted category/payee/account autocomplete modals from EditField to their own files/components

This list is a great starting point for the breaking-up into smaller chunks. Especially the bigger changes such as file/component moves (with no logic changes) could be extracted to separate PRs for a big drop in LOC here.

joel-jeremy commented:
@MatissJanis I've split the autocomplete changes here: https://github.com/actualbudget/actual/pull/2500

joel-jeremy commented:
Split menu components here: https://github.com/actualbudget/actual/pull/2511

joel-jeremy commented:
@MatissJanis This should now be ready for review :) Most related refactors have been merged to master.

MatissJanis commented on packages/desktop-client/src/components/modals/CoverModal.tsx:
```diff
@@ -0,0 +1,115 @@
+
+  const category = categories.find(c => c.id === categoryId);
+
+  if (category == null) {
```
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** 

```suggestion
  if (category === null) {
```

MatissJanis reviewed (commented):
Overall LGTM. Left a few comments. Let me know what you think.
